web/docs/pipelines/README.md

```diff
@@ -0,0 +1,231 @@
+In this case the operator transparently "demultiplex" a heterogeneous event
+stream into N homogeneous streams. The [`import`](tql2/operators/import.md)
+operator is an example of such an operator; it pushes its ordering upstream,
+allowing parsers to efficiently create multiple streams events.
+
+![Multi-schema Example](multi-schema-example.svg)
+
+Some operators only work with exactly one instance per schema internally, such
+as [`write_csv`](tql2/operators/write_parquet.md), which first writes a header
+and then all subsequent rows have to adhere to the emitted schema. Such
+operators cannot handle events with changing schemas.
+
+It's important to mention that most of the time you don't have to worry about
+schemas. They are there for you when you want to work with them, but it's often
+enough to just specified the fields that you want to work with, e.g., `where
+id.orig_h in 10.0.0.0/8`, or `select src_ip, dest_ip, proto`. Schemas are
+inferred automatically in parsers, but you can also seed a parser with a schema
+that you define explicitly.
+
+## Unified Live Stream Processing and Historical Queries
+
+Engines for event stream processing and batch processing of historical data have
+vastly different requirements. We believe that we found a sweetspot with our
+language and accompanying execution engine that makes working with both types of
+workloads incredibly easy: just pick a input operator at the beginning the a
+pipeline that points to your data source, be it infinitely streaming or stored
+dataset. Tenzir will figure out the rest.
+
+![Unified Processing](unified-processing.svg)
+
+Our desired user experience for interacting with historical data looks like
+this:
+
+1. **Ingest**: to store data at a node, create a pipeline that ends with
+   [`import`](tql2/operators/import.md).
+2. **Query**: to run a historical query over data at the node, create a pipeline
+   that begins with [`export`](tql2/operators/export.md).
+
+For example, to ingest JSON from a Kafka, you write `from "kafka://topic |
+import`. To query the stored data, you write `export | where file == 42`.
+
+The example with `export` suggests that the pipeline *first* exports everything,
+and only *then* starts filtering with `where`, performing a full scan over the
+stored data. But this is not what's happening. Pipelines support **predicate
+pushdown** for every operator. This means that `export` receives the filter
+expression before it starts executing, enabling index lookups or other
+optimizations to efficiently execute queries with high selectivity where scans
+would be sub-optimal.
+
+The key insight here is to realize that optimizations like predicate pushdown
+extend to the storage engine and do not only apply to the streaming executor.
+
+The Tenzir native storage engine is not a full-fledged database, but rather a
+catalog a thin indexing layer over a set of Parquet/Feather files. The sparse
```

Comment:
"a catalog a thin indexing layer" should likely be either "a catalog" or "a thin indexing layer."